pkgs/development/python-modules/xlib/default.nix

```
@@ -3,24 +3,32 @@
 , fetchFromGitHub
 , six
 , setuptools_scm
-, pkgs
+, xorg
+, mock
+, nose
+, utillinux
 }:
 
 buildPythonPackage rec {
   pname = "xlib";
-  version = "0.17";
+  version = "0.25";
 
   src = fetchFromGitHub {
     owner = "python-xlib";
     repo = "python-xlib";
-    rev = "${version}";
-    sha256 = "1iiz2nq2hq9x6laavngvfngnmxbgnwh54wdbq6ncx4va7v98liyi";
+    rev = version;
+    sha256 = "1nncx7v9chmgh56afg6dklz3479s5zg3kq91mzh4mj512y0skyki";
   };
 
-  # Tests require `pyutil' so disable them to avoid circular references.
-  doCheck = false;
+  doCheck = true;
 
-  propagatedBuildInputs = [ six setuptools_scm pkgs.xorg.libX11 ];
+  checkPhase = ''
+    python runtests.py
```

```suggestion
    ${python.interpreter} runtests.py
```
That's needed because e.g. PyPy has a different executable name.